### PR TITLE
Symfony 6.3 and API Platform 2.7

### PR DESCRIPTION
**Pull request by rimi-itk**
#### Link to ticket

https://jira.itkdev.dk/browse/DISPLAY-1030

#### Description

Updates to Symfony 6.3 and API Platform 2.7 (cf. https://api-platform.com/docs/core/upgrade-guide/#im-migrating-from-26-and-want-to-prepare-for-30)

#### Checklist

- [x] My code is covered by test cases.
- [x] My code passes our test (all our tests).
- [x] My code passes our static analysis suite.
- [x] My code passes our continuous integration process.

The API Specification has some (hopefully) insignificant changes:

* Empty descriptions added
* Generic name "Resource" changed to actual resource name in descriptions, e.g. "ScreenGroupCampaign identifier" vs. "Resource identifier"
* `links` are not set in response examples (do we use these links?) This is related to the line https://github.com/api-platform/core/blob/v2.6.8/src/OpenApi/Factory/OpenApiFactory.php#L188 (and others) having been removed and refactored extensively.
* `examples` and additional properties added in `schema`

https://gist.github.com/rimi-itk/ff2c9400aa8d6d76caae698053b54638#file-compare-api-specs-sh has been used to test the API spec piece by piece and compare it with https://github.com/os2display/display-api-service/tree/develop

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -72,12 +73,20 @@ services:
     labels:
       - "traefik.enable=true"
       - "traefik.docker.network=frontend"
-      - "traefik.http.routers.${COMPOSE_PROJECT_NAME}Mailhog.rule=Host(`mailhog-${COMPOSE_DOMAIN}`)"
-      - "traefik.http.services.${COMPOSE_PROJECT_NAME}Mailhog.loadbalancer.server.port=8025"
+      - "traefik.http.routers.${COMPOSE_PROJECT_NAME}mail.rule=Host(`mail-${COMPOSE_DOMAIN}`)"
+      - "traefik.http.services.${COMPOSE_PROJECT_NAME}mail.loadbalancer.server.port=8025"
 
   redis:
```
Comment (tuj) [inline docker-compose.yml:79]: Should we move redis and node to docker-compose.override.yml ?

File: migrations/Version20231006120901.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
+<?php
```
Comment (tuj) [inline migrations/Version20231006120901.php:1]: This seems like it should be removed.
Why was it added?

File: src/Serializer/RelationNormalizer.php [hunk 3]
```diff
@@ -67,6 +59,7 @@ public function normalize($object, $format = null, array $context = []): array|s
         $data = ['@context' => '/contexts/'.$context['output']['name']];
 
         if (isset($context['operation_type']) && OperationType::SUBRESOURCE === $context['operation_type']) {
+            // FIXME: When do we end up here? `getSubresourceIriFromResourceClass` is not defined anymore.
```
Comment (tuj) [inline src/Serializer/RelationNormalizer.php:62]: Should this be removed?

File: src/Serializer/RelationNormalizer.php [hunk 4]
```diff
@@ -93,9 +86,18 @@ public function normalize($object, $format = null, array $context = []): array|s
     /**
      * {@inheritdoc}
      */
-    public function hasCacheableSupportsMethod(): bool
+    public function getSupportedTypes(?string $format)
     {
-        return true;
+        // FIXME: Figure out what to actually return here?
```
Comment (tuj) [inline src/Serializer/RelationNormalizer.php:91]: Is this handled?

File: tests/Api/ScreenGroupsTest.php [hunk 5]
```diff
@@ -163,7 +164,9 @@ public function testGetScreenGroupsScreenRelations(): void
         $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
         $this->assertJsonContains([
             '@context' => '/contexts/ScreenGroup',
-            '@id' => '/v1/screen-groups',
+            // FIXME: What's the correct `@id` value?
```
Comment (tuj) [inline tests/Api/ScreenGroupsTest.php:167]: Remove this and next line. "'@id' => '/v1/screens/'.$ulid.'/screen-groups'," looks correct.

**Review by tuj — CHANGES_REQUESTED** [review]
Generally looks good.

File: migrations/Version20231006120901.php (hunk 2 above)
Comment (rimi-itk) [reply migrations/Version20231006120901.php:1]: It was added in https://github.com/os2display/display-api-service/pull/165/commits/30ca376725da339545a76db51ef9cf07d9bbcdc5 to make our “Validate Doctrine Schema” stop complaining. It sure is weird …

File: TODO.md
[the remark this answers is not in the document]
Comment (rimi-itk) [reply TODO.md:1]: Done in https://github.com/os2display/display-api-service/pull/165/commits/757cdbd81914e0a705b60c61c66c1cc45a4297ee

File: tests/Api/ScreenGroupsTest.php (hunk 5 above)
Comment (rimi-itk) [reply tests/Api/ScreenGroupsTest.php]: Removed in https://github.com/os2display/display-api-service/pull/165/commits/757cdbd81914e0a705b60c61c66c1cc45a4297ee

File: docker-compose.yml (hunk 1 above)
Comment (rimi-itk) [reply docker-compose.yml:79]: That would be the ITK Dev way to do it, but https://github.com/os2display/display-api-service/commit/04cbc36771a7b7020b148ffa035876223571c79d

File: migrations/Version20231006120901.php (hunk 2 above)
Comment (rimi-itk) [reply migrations/Version20231006120901.php:1]: A comment on this weird stuff added in https://github.com/os2display/display-api-service/pull/165/commits/d453c118d03a6b1c8b7b64d1faba7e6b1897d5cf

**Review by tuj — APPROVED** [review]

**Pull request merged**